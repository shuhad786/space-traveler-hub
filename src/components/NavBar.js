import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => (
  <nav className="navBar">
    <div className="brandNameBox">
      <img className="navLogo" src="planet.png" alt="logo space" />
      <h1 className="navTitle">Space Travelers Hub</h1>
    </div>
    <ul className="links">
      <NavLink className="link" to="/">Rockets</NavLink>
      <NavLink className="link" to="/missions">Missions</NavLink>
      <div className="navVertLine" />
      <NavLink className="link" to="/my-profile">My Profile</NavLink>
    </ul>
  </nav>
);

export default Navbar;
