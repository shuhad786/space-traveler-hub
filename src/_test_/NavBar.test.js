import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import NavBar from '../components/NavBar';

describe('Navbar', () => {
  test('render navbar correctly', () => {
    const navbar = render(
      <React.StrictMode>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </React.StrictMode>,
    );
    expect(navbar).toMatchSnapshot();
  });
});
