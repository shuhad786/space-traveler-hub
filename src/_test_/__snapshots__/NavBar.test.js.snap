// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar render navbar correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navBar"
      >
        <div
          class="brandNameBox"
        >
          <img
            alt="logo space"
            class="navLogo"
            src="planet.png"
          />
          <h1
            class="navTitle"
          >
            Space Travelers Hub
          </h1>
        </div>
        <ul
          class="links"
        >
          <a
            aria-current="page"
            class="link active"
            href="/"
          >
            Rockets
          </a>
          <a
            class="link"
            href="/missions"
          >
            Missions
          </a>
          <div
            class="navVertLine"
          />
          <a
            class="link"
            href="/my-profile"
          >
            My Profile
          </a>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navBar"
    >
      <div
        class="brandNameBox"
      >
        <img
          alt="logo space"
          class="navLogo"
          src="planet.png"
        />
        <h1
          class="navTitle"
        >
          Space Travelers Hub
        </h1>
      </div>
      <ul
        class="links"
      >
        <a
          aria-current="page"
          class="link active"
          href="/"
        >
          Rockets
        </a>
        <a
          class="link"
          href="/missions"
        >
          Missions
        </a>
        <div
          class="navVertLine"
        />
        <a
          class="link"
          href="/my-profile"
        >
          My Profile
        </a>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
